server/aesdsocket.c

```diff
@@ -1,331 +1,580 @@
+    else if(client_address.ss_family == AF_INET6){
+        char addr[INET6_ADDRSTRLEN];
+        struct sockaddr_in6 *addr_in6 = (struct sockaddr_in6 *)&client_address;
+        inet_ntop(AF_INET6, &(addr_in6->sin6_addr), addr, INET6_ADDRSTRLEN);
+        syslog(LOG_INFO, "Closed connection from %s", addr);
+    }
+}
+
+//Handle a client that we accepeted from a socket connection
+void *socket_handle1(void *thread_info)
+{
+    struct thread_t *thread_data = (struct thread_t *) thread_info;
+
+    //Print IP
+    accept_connection(thread_data->client_address);
 
-    //Wait for data
     int recv_ret;
-    //Start with a size, potentially incresing it if an entire packet cannot fit into it
-    recv_data = malloc(sizeof(char) * BUFFER_SIZE);
+    int index = 0;
+    //Keep track of number of BUFFER_SIZE receive_blocks "recv_data" has
+    int receive_blocks = 1;
+    
+    char *recv_data = malloc(sizeof(char)*BUFFER_SIZE*receive_blocks);
     if(!recv_data){
-    	syslog(LOG_ERR, "Error occured during malloc = %s. Exiting...", strerror(errno));
-    	exit(-1);
+        syslog(LOG_ERR, "Error occured during malloc() in recv = %s. Exiting...", strerror(errno));
+        goto exit_label2; 
     }
-    int recv_idx = 0;
-    //How many 'BUFFER_SIZE' receive_blocks "recv_data" we have received from client till now
-    int receive_blocks = 1;
-    do {
-      recv_ret = recv(conn_fd, & recv_data[recv_idx], BUFFER_SIZE, 0);
-      if (recv_ret == -1) {
-        syslog(LOG_ERR, "Error occured while reading from socket = %s. Exiting...", strerror(errno));
-        exit(-1);
-      }
-//      printf("Received data = %s\n", &recv_data[recv_idx]);
-      recv_idx += recv_ret;
 
-      if (recv_idx != 0) {
-        if (recv_data[recv_idx - 1] == '\n') {
-
-          if (lseek(log_fd, 0, SEEK_END) == -1) {
-            syslog(LOG_ERR, "Error occured while lseek() to EOF = %s. Exiting...", strerror(errno));
-            exit(-1);
-          }
-
-          int write_byte_ret;
-          int write_len = recv_idx;
-          char * write_ptr = recv_data;
-          while (write_len != 0) {
-            write_byte_ret = write(log_fd, write_ptr, write_len);
-            
-            if (write_byte_ret == -1) {
-              if (errno == EINTR)
-                continue;
-              else {
-                syslog(LOG_ERR, "Error occured while writing to file = %s. Exiting...", strerror(errno));
-                exit(-1);
-              }
+    //Receive data from client and store in file continuously till EOP is received
+    do{
+        recv_ret = recv(thread_data->client_socket_t, &recv_data[index], BUFFER_SIZE, 0);
+        if(recv_ret == -1){
+            syslog(LOG_ERR, "Error occured during recv() while reading from socket = %s. Exiting...", strerror(errno));
+            goto exit_label3;  
+        }
+        index += recv_ret;
+
+        if(index != 0){
+            //Check if EOP "\n" is received
+            if(recv_data[index - 1] == '\n'){
```

Comment:
@prvbharadwaj1731 what happens if the recv packet does not end in `\n`?  See previous review comment at https://github.com/cu-ecen-aeld/assignments-3-and-later-prvbharadwaj1731/pull/3#discussion_r987546910